internal/exec/stages/disks/luks.go

```diff
@@ -140,6 +146,66 @@ func (s *stage) createLuks(config types.Config) error {
 				}
 			}
 		}
+
+		if luks.WipeVolume == nil || !*luks.WipeVolume {
+			// If the volume isn't forcefully being created, then we need
+			// to check if it is of the correct type or that no volume exists.
+
+			// check if the LUKS device already exists, device will be
+			// opened if it exists; don't allow clevis device re-use as
+			// we can't guarantee that what is stored in the header is
+			// exactly what would be generated from the given config
+			var exists bool
+			if !ignitionCreatedKeyFile && luks.Clevis == nil {
+				var err error
+				exists, err = s.checkLuksDeviceExists(luks, keyFilePath)
+				if err != nil {
+					return fmt.Errorf("checking if luks device exists: %v", err)
+				}
+			}
+
+			var info execUtil.FilesystemInfo
+			err := s.Logger.LogOp(
+				func() error {
+					var err error
+					info, err = execUtil.GetFilesystemInfo(devAlias, false)
+					if err != nil {
+						// Try again, allowing multiple filesystem
+						// fingerprints this time.  If successful,
+						// log a warning and continue.
+						var err2 error
+						info, err2 = execUtil.GetFilesystemInfo(devAlias, true)
+						if err2 == nil {
+							s.Logger.Warning("%v", err)
+						}
+						err = err2
+					}
+					return err
+				},
+				"determining volume type of %q", *luks.Device,
+			)
+			if err != nil {
+				return err
+			}
+			s.Logger.Info("found %s at %q with uuid %q and label %q", info.Type, *luks.Device, info.UUID, info.Label)
+
+			if exists {
+				s.Logger.Info("volume at %q is already correctly formatted. Skipping...", *luks.Device)
+				continue
+			} else if info.Type != "" {
+				s.Logger.Err("volume at %q is not of the correct type, label, or UUID (found %s, %q, %s) and a volume wipe was not requested", *luks.Device, info.Type, info.Label, info.UUID)
+				return ErrBadVolume
+			}
+		}
+
+		if _, err := s.Logger.LogCmd(
+			exec.Command(distro.WipefsCmd(), "-a", devAlias),
```

Comment:
Hmm, a doubt here though I might be misunderstanding the flow. In the re-use path, shouldn't we be skipping all of this up to an including `luksOpen` since we already opened it in `checkLuksDeviceExists`?

Reply:
In the re-use path `wipeVolume` must be `false` and if the device matches we `continue` to go to the next LUKS entry skipping over this path entirely.

Reply:
For clarity, the wipefs could be put in an `else` of the `WipeVolume` check.  If we got here with `WipeVolume` false, we already know there's no FS signature.